java-frontend/src/main/java/org/sonar/java/resolve/Resolve.java

```diff
@@ -96,13 +96,16 @@ private JavaType substituteTypeParameter(JavaType type, TypeSubstitution substit
         newSubstitution.add(entry.getKey(), substituteTypeParameter(entry.getValue(), substitution));
       }
       return parametrizedTypeCache.getParametrizedTypeType(ptt.rawType.getSymbol(), newSubstitution);
-    }
-    if (type instanceof JavaType.WildCardType) {
+    } else if (type instanceof JavaType.WildCardType) {
       JavaType.WildCardType wildCardType = (JavaType.WildCardType) type;
       substitutedType = substitution.substitutedType(wildCardType.bound);
       if (substitutedType != null) {
         return parametrizedTypeCache.getWildcardType(substitutedType, wildCardType.boundType);
       }
+    } else if (type instanceof JavaType.ArrayJavaType) {
+      JavaType elementType = ((JavaType.ArrayJavaType) type).elementType;
+      // FIXME SONARJAVA-1574
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1134)